packages/flutter/lib/src/gestures/gesture_settings.dart

```diff
@@ -0,0 +1,52 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui;
+import 'package:flutter/foundation.dart';
+
+/// The device specific gesture settings scaled into logical pixels.
+///
+/// This configuration can be retrieved from the window, or more commonly from a
+/// [MediaQuery] widget.
+///
+/// See also:
+///
+///  * [ui.GestureSettings], the configuration that this is derived from.
+@immutable
+class DeviceGestureSettings {
+  /// Create a new [DeviceGestureSettings] with configured settings in logical
+  /// pixels.
+  const DeviceGestureSettings({
+    this.touchSlop,
+  });
+
+  /// Create a new [DeviceGestureSettings] from the provided [window].
+  factory DeviceGestureSettings.fromWindow(ui.SingletonFlutterWindow window) {
+    final double? physicalTouchSlop = window.viewConfiguration.gestureSettings.physicalTouchSlop;
+    return DeviceGestureSettings(
+      touchSlop: physicalTouchSlop == null ? null : physicalTouchSlop / window.devicePixelRatio
+    );
+  }
+
+  /// The touch slop value in logical pixels, or `null` if it was not set.
+  final double? touchSlop;
+
+  /// The touch slop value for pan gestures, in logical pixels, or `null` if it
+  /// was not set.
```

Comment:
Looks like there is no way to actually set it independently? Maybe document that it's either null or 2* touchSlop if touchSlop is set?

Reply:
Will do. In the future we could support setting it separately if need be.